presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -551,7 +552,34 @@ private PlanBuilder aggregate(PlanBuilder subPlan, QuerySpecification node)
         if (needPostProjectionCoercion) {
             return explicitCoercionFields(subPlan, distinctGroupingColumns, analysis.getAggregates(node));
         }
-        return subPlan;
+
+        // 4. Project and re-write all grouping functions
+        return handleGroupingOperations(subPlan, node, groupIdSymbol);
+    }
+
+    private PlanBuilder handleGroupingOperations(PlanBuilder subPlan, QuerySpecification node, Optional<Symbol> groupIdSymbol)
+    {
+        if (analysis.getGroupingOperations(node).isEmpty()) {
+            return subPlan;
+        }
+
+        TranslationMap translations = subPlan.copyTranslations();
+
+        Assignments.Builder projections = Assignments.builder();
+        projections.putIdentities(subPlan.getRoot().getOutputSymbols());
+
+        for (GroupingOperation groupingOperation : analysis.getGroupingOperations(node)) {
```

Comment:
the rewrite here doesn't account for coercions, e.g:
```
select grouping(a) * 1.0 from ... group by a
```
should fail (add test).

I think the loop should be:
```
Expression rewritten = GroupingOperationRewriter.rewriteGroupingOperation(groupingOperation, node, analysis, metadata, groupIdSymbol);
Type coercion = analysis.getCoercion(groupingOperation);
Symbol symbol = symbolAllocator.newSymbol(rewritten, analysis.getTypeWithCoercions(groupingOperation));
if (coercion != null) {
    rewritten = new Cast(
        rewritten,
        coercion.getTypeSignature().toString(),
        false,
        metadata.getTypeManager().isTypeOnlyCoercion(type, coercion));
}
projections.put(symbol, rewritten);
translations.put(rewritten, symbol);
```